src/util.js

```diff
@@ -310,23 +310,24 @@ function heavyCompare (x, y, g) {
     lis.sort()
     return lis.join('\n')
   }
-  if ((x.termType === 'BlankNode') || (y.termType === 'BlankNode')) {
+  if ((x.termType === 'BlankNode') && (y.termType === 'BlankNode')) {
     if (x.compareTerm(y) === 0) return 0 // Same
     if (signature(x) > signature(y)) return +1
     if (signature(x) < signature(y)) return -1
     return x.compareTerm(y)  // Too bad -- this order not canonical.
     // throw "different bnodes indistinquishable for sorting"
   } else {
+    if (uriMap && x.uri && y.uri){
+      return (uriMap[x.uri] || x.uri).localeCompare(uriMap[y.uri] || y.uri)
```

Comment:
The condition `x.uri && y.uri `on L320 filters out any condition except when we are comparing two things with a `.uri`, i.e. `NamedNode` objects